daemon/api_model.go

```diff
@@ -69,3 +89,90 @@ func postModel(c *Command, r *http.Request, _ *auth.UserState) Response {
 	return AsyncResponse(nil, &Meta{Change: chg.ID()})
 
 }
+
+// helper for parsing url options into formatting option vars
+func parseHeadersFormatOptionsFromURL(url *url.URL) (
```

Comment:
this function can be improved significantly by:
* have it take a `url.Values` instead of the `url` directly, making it more obvious it's getting it from the Query and not the whole URL
* returning an ad-hoc struct instead of a bag of values that are hard to track (i.e. and e.g. `(modelOptions, error)`)

Furthermore, it _looks_ like all you're doing is checking `url.Query()["json"]`, for which the function looks like overkill.

Reply:
This was modified from the api_asserts.go formatting option parsing, I was going to also implement using the helper there too where things like `headers` is used, but seem to have forgotten about using it there too. I have refactored as per your suggestions